tests/conftest.py

```diff
@@ -46,3 +57,28 @@ def testbed(request):
 
     tbinfo = TestbedInfo(tbfile)
     return tbinfo.testbed_topo[tbname]
+
+@pytest.fixture(autouse=True)
+def loganalyzer(duthost, request):
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="loganalyzer")
+    # Add start marker into DUT syslog
+    marker = loganalyzer.init()
+    yield loganalyzer
+    if not request.config.getoption("--disable_loganalyzer") and not pytest.disable_loganalyzer:
+        # Read existed common regular expressions located with legacy loganalyzer module
+        loganalyzer.load_common_config()
+        result = loganalyzer.analyze(marker)
+        if not result:
+            pytest.fail("Log analyzer failed.")
+        if result["total"]["match"] != 0:
+            pytest.fail("Found errors: {}".format(result))
+    else:
+        # Add end marker into DUT syslog
+        loganalyzer._add_end_marker(marker)
```

Comment:
If we fail in above condition why don't we put end marker in any case ?

Reply:
End marker will be inserted before fail by "analyze" method.